File: Assignment_2/CPP_Builder/PizzaEmployee.cpp
#include "PizzaEmployee.h"

PizzaEmployee::PizzaEmployee(PizzaBuilder* pizza_builder) : pizzaBuilder (pizza_builder)
{
}

void PizzaEmployee::MakePizza(PizzaSize size, std::vector<PizzaTopping> toppings)
{
    pizzaBuilder->MakePizza(size, toppings);
}

std::unique_ptr<Pizza> PizzaEmployee::GetPizza()
{
    return std::move(pizzaBuilder->GetPizza());
}

File: Assignment_2/JAVA_Builder/Builder/src/PizzaEmployee.java
import java.util.Vector;

class PizzaEmployee {
    private PizzaBuilder pizzaBuilder = null;

    PizzaEmployee(PizzaBuilder pizza_builder) {
        pizzaBuilder = pizza_builder;
    }

    public void MakePizza(PizzaSize size, Vector<PizzaTopping> toppings) {
        pizzaBuilder.MakePizza(size, toppings);
    }

    public Pizza GetPizza() {
        return pizzaBuilder.GetPizza();
    }
};
